src/components/Form/Amount/index.tsx

```diff
@@ -36,71 +36,62 @@ const Amount = ({
-                if (e.code === 'Minus' || e.code === 'KeyE') {
-                  e.preventDefault();
-                }
-              }}
-              placeholder="0.0"
-              {...register}
-              value={defaultValue}
-              readOnly={readOnly}
-            />
-          </div>
-          <div className="flex">
-            {hideHalfButton ? (
-              <></>
-            ) : (
-              <>
-                <button
-                  className="text-accent-content underline hover:opacity-70 mx-1 font-semibold"
-                  onClick={() => setValue(roundNumber(Number(max) * 0.5))}
-                  type="button"
-                >
-                  50%
-                </button>
-              </>
-            )}
-            {hideMaxButton ? (
-              <></>
-            ) : (
-              <>
-                <button
-                  className="text-accent-content underline hover:opacity-70 mx-1 font-semibold"
-                  onClick={() => setValue(calculateMaxAmount(Number(max), fullMax))}
-                  type="button"
-                >
-                  MAX
-                </button>
-              </>
-            )}
-          </div>
+}: AmountProps): JSX.Element | null => (
+  <>
+    <div
+      className={`rounded-lg bg-base-300 px-4 py-3 ${className || ''} ${
+        error ? 'border border-solid border-red-400' : ''
+      }`}
+    >
+      <div className="w-full flex justify-between">
+        <div className="flex-grow text-4xl text-accent-content font-outfit">
+          <NumericInput
+            register={register}
+            readOnly={readOnly}
+            additionalStyle="input-ghost w-full text-4xl font-outfit pl-0 focus:outline-none"
+            defaultValue={defaultValue}
+          />
         </div>
-        <div className="flex justify-between items-center dark:text-neutral-400 text-neutral-500">
-          <div className="text-sm mt-px">Amount</div>
-          <div className="flex gap-1 text-sm">
-            {max !== undefined && <span className="mr-1">Available: {max.toFixed(2)}</span>}
-          </div>
+        <div className="flex">
+          {hideHalfButton ? (
+            <></>
+          ) : (
+            <>
+              <button
+                className="text-accent-content underline hover:opacity-70 mx-1 font-semibold"
+                onClick={() => setValue(roundNumber(Number(max) * 0.5))}
+                type="button"
+              >
+                50%
+              </button>
+            </>
+          )}
+          {hideMaxButton ? (
+            <></>
+          ) : (
+            <>
+              <button
+                className="text-accent-content underline hover:opacity-70 mx-1 font-semibold"
+                onClick={() => setValue(calculateMaxAmount(Number(max), fullMax))}
+                type="button"
+              >
+                MAX
+              </button>
+            </>
+          )}
```

Comment:
Nit: Using `&&` and removing the wrapping `<></>` might make it easier to read or do we need `<></>`?
```suggestion
        <div className="flex">
          {hideHalfButton && (
              <button
                className="text-accent-content underline hover:opacity-70 mx-1 font-semibold"
                onClick={() => setValue(roundNumber(Number(max) * 0.5))}
                type="button"
              >
                50%
              </button>
          )}
          {hideMaxButton && (
              <button
                className="text-accent-content underline hover:opacity-70 mx-1 font-semibold"
                onClick={() => setValue(calculateMaxAmount(Number(max), fullMax))}
                type="button"
              >
                MAX
              </button>
          )}
```

Reply:
@ebma https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx

Reply:
Well, today I learned 😂

Reply:
But you're right we can simplify it this way:

` <div className="flex">
          {hideHalfButton ? null : (
            <button
              className="text-accent-content underline hover:opacity-70 mx-1 font-semibold"
              onClick={() => setValue(roundNumber(Number(max) * 0.5))}
              type="button"
            >
              50%
            </button>
          )}
          {hideMaxButton ? null : (
            <button
              className="text-accent-content underline hover:opacity-70 mx-1 font-semibold"
              onClick={() => setValue(calculateMaxAmount(Number(max), fullMax))}
              type="button"
            >
              MAX
            </button>
          )}
   </div>
`